src/spike_psvae/chunk_features.py

```diff
@@ -76,7 +76,7 @@ def handle_which_wfs(self, subtracted_wfs, cleaned_wfs, denoised_wfs):
                 f"which_waveforms={self.which_waveforms} not in ('subtracted', 'cleaned', denoised)"
             )
         if not self.tensor_ok and torch.is_tensor(wfs):
-            wfs = wfs.cpu().numpy()
+            wfs = wfs.cpu().detach().numpy()
```

Comment:
you may want to make sure that you have requires_grad=False on your waveforms and model (.eval()). (you can also call torch.set_grad_enabled(False)) --- detach is only necessary when gradients are being used